src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerProvider.Windows.cs

```diff
@@ -0,0 +1,25 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Extensions.Logging.Console
+{
+    /// <summary>
+    /// A provider of <see cref="ConsoleLogger"/> instances.
+    /// </summary>
+    public partial class ConsoleLoggerProvider : ILoggerProvider, ISupportExternalScope
+    {
+        private static bool DoesConsoleSupportAnsi()
+        {
+            // for Windows, check the console mode
```

Comment:
Do we also want to check if the output is redirected? (maybe if we do, we probably want that on both windows and non-windows)

Reply:
It is not necessary if the output is directed will not be used in display which the escape code can still be needed. This is hard to detect the intention of the redirection.

Reply:
FYI: https://github.com/dotnet/runtime/issues/37421
The existing behavior discussed in the above issue is because the code in master embeds the codes in AnsiLogConsole irrespective of the value of DisableColors.

As-is currently with the PR, if output is redirected then when DisableColors is true, then the bug won't reproduce anymore.
Otherwise when DisableColors is false, then I don't think it matters which console mode we are using because the ansi codes would have already been written by then formatter, nothing to do with which console mode is being used.

I think the question would need to be, do we embed ANSI codes in the formatter when console output is redirected or not. Is guess since the DisableColors property is available no need to do any extra check. If anything it is also possible the redirected console could also support ANSI output and we would be disabling colors there.